### PR TITLE
Added support for ViewPager2

### PR DESCRIPTION
**Pull request by CaptnBlubber**
This will add Support for ViewPager2 wich internally works with a recyclerview. Therefore the same pattern as with Recyclerview can be easily applied here.

There is one pitfall with ViewPager2: When used in combination with a TabLayout changing the adapter will not notify the TabLayout about a new adapter and therefore the tab amount wont be reflecting the current active adapter.

For most cases I think this is an okay behavior as for viewpagers you probably dont want to have swipeable loading content, therefore in that case we want to have 1 item populated in the Skeleton adapter anyways. The Tablayout will remain attached to the original adapter and display nothing, once the original adapter gets populated and the skeleton is hidden the Tablayout will reflect the correct amount of items.

If however the TabLayout should reflect the correct amount of Skeleton Views and real views the Tablayout needs to reattach to the correct adapter everytime a toggle happens. I have included the workaround solution in the example app like this:
```
override fun onSkeletonToggled() {
        //We are swapping out the adapter when showing/hiding the skeleton.
        //Therefore we need to reattach the mediator in order to show the correct amount of Tabs
        mediator.detach()
        mediator.attach()
    }
```

This could be automated within the utils however this would imply having another dependency in the library to material-components wich i think is unnecessary for that edgecase handling. Happy to discuss on that one

### CODE REVIEW

**Comment by Faltenreich** [comment]
Thank you very much for your effort!

I have one question before I am considering merging this pull request: is there really a use case for masking the content of a ViewPager2 instead of its children one by one? Most ViewPagers show content with different data and therefore the asynchronous operation for one page may have been finished but not for the other ones. This use case would not be covered when masking the whole ViewPager2, since either all children can be masked or none.

Your workaround also shows the first problem when masking a ViewPager2 and I am not sure if there may be more problems hidden beneath. Therefore I would rather advice applying the Skeleton to every single page that should be masked.

If there is something I am missing, I will reconsider this pull request anyhow, since it is always upon the user of the library how to use a library and this extension seems not intrusive at all.

**Comment by CaptnBlubber** [comment]
There are not only Fragment based implementations of ViewPager2 contents that handle data loading individually.

If you have a content list that should only scroll one by one element when swiping/flinging you could extend recyclerview to modify snap behavior or just use ViewPager2.

I agree that with fragments handling their own data its better to use the Skeleton wrapper inside the layout.

**Comment by CaptnBlubber** [comment]
Let me try to picture an exemplary Implementation where it might make sense:

Lets pretend we are building a Shopping Application. 
There is a pretty popular one starting with an A that has a name similar to a giant River that has exactly this behavior ;)

We have a ProductDetail Fragment that displays all relevant pricing informations and Product Pictures.
The Pictures should behave like viewpager, cycling one by one.

The API returns all product pricing informations and product images in one API call.

While the Fragment loads the one API request the Skeleton should appear. Therefore we mask all TextViews like the price and product name and we need to mask the space that is taken by the ProductImages.

**Comment by Faltenreich** [comment]
Got it! I reformatted parts of the pull request and fixed a crash on API level 20 and lower. Thank you very much for your pull request! It will be merged into master and become a part of the library in the upcoming version 3.0.0.
